TLM/TLM/UI/ModUI.cs

```diff
@@ -113,11 +104,13 @@ public class UIOpacityObservable : GenericObservable<UIOpacityNotification> {
             }
         }
 
-        ~ModUI() {
+        public void Destroy() {
             Log._Debug("ModUI destructor is called.");
             Destroy(MainMenuButton);
             Destroy(MainMenu);
             ReleaseTool();
+            Instance = null;
+            Destroy(this);
```

Comment:
Doesn't `this` get destroyed anyway?

Reply:
No, check the changes I made to LoadingExtension.cs.
Before it would call 
Object.Destroy(ModUI.Instance);
then do the code above in the Finalizer.
Since I removed the Finalizer, or rather made it the Destroy() method, I call that from LoadingExtension instead.
Now it still has to destroy itself.